src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final double GROW_VALUE = 1.5;
+    private int maxSize = 10;
+    private int actualSize = 0;
+    private Object[] array = new Object[maxSize];
+    private Object[] tempArray;
+
     @Override
     public void add(T value) {
-
+        if (actualSize == maxSize) {
+            grow();
+        }
+        array[actualSize] = value;
+        actualSize++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > actualSize || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't exist");
+        }
+        if (actualSize == maxSize) {
+            grow();
+        }
+        tempArray = new Object[maxSize];
+        System.arraycopy(array, 0, tempArray, 0, index);
+        tempArray[index] = value;
+        System.arraycopy(array, index, tempArray, index + 1, actualSize - index);
+        actualSize++;
+        array = tempArray;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int requiredSize = maxSize - actualSize + list.size();
+        while (requiredSize > maxSize) {
+            grow();
+        }
+        tempArray = new Object[maxSize];
+        System.arraycopy(array, 0, tempArray, 0, actualSize);
+        System.arraycopy(toArray(list), 0, tempArray, actualSize, list.size());
+        actualSize = actualSize + list.size();
+        array = tempArray;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= actualSize || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't exist");
```

Comment:
include index and size . extract this checker to a private utility method to avoid code duplication